jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -79,11 +83,13 @@ def parse_data(app, file_obj, data_type=None, data_label=None):
     # If the data types are custom data objects, use explicit parsers. Note
     #  that this relies on the glue-astronomy machinery to turn the data object
     #  into something glue can understand.
-    elif isinstance(file_obj, Spectrum1D):
+    elif isinstance(file_obj, Spectrum1D) and file_obj.flux.ndim in (1, 3):
```

Comment:
Allowing `ndim=2` being parsed as if it is `ndim=1` is a bug but since no one complained about it, I didn't bother with a change log.